src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -172,18 +173,53 @@
     private String serverName;
     private String gerritSlaveId;
     private List<PluginGerritEvent> triggerOnEvents;
-    @SuppressWarnings("unused")
-    @Deprecated
-    private transient boolean allowTriggeringUnreviewedPatches;
     private boolean dynamicTriggerConfiguration;
     private String triggerConfigURL;
 
     private GerritTriggerTimerTask gerritTriggerTimerTask;
-
     private GerritTriggerInformationAction triggerInformationAction;
 
     /**
      * Default DataBound Constructor.
+     * @param gerritProjects the set of triggering rules.
+     */
+    @DataBoundConstructor
+    public GerritTrigger(List<GerritProject> gerritProjects) {
+        this.gerritProjects = gerritProjects;
+        this.gerritTriggerTimerTask = null;
+        this.triggerInformationAction = new GerritTriggerInformationAction();
+        this.skipVote = new SkipVote(false, false, false, false);
+        this.escapeQuotes = true;
+        this.serverName = ANY_SERVER;
+        try {
+            DescriptorImpl descriptor = (DescriptorImpl)getDescriptor();
+            if (descriptor != null) {
+                ListBoxModel options = descriptor.doFillNotificationLevelItems(this.serverName);
+                if (!options.isEmpty()) {
+                    this.notificationLevel = options.get(0).value;
+                }
+            }
+            //CS IGNORE EmptyBlock FOR NEXT 1 LINES. REASON: Handled one row below
+        } catch (NullPointerException ignored) { /*Could happen during testing*/ }
+        if (this.notificationLevel == null) {
```

Comment:
They are also set in the field definition, this is just some extra security to get sensible defaults in case the object is created outside of form posting.